### DISCUSSION TITLE
ERROR: tflite_runtime-2.6.0-cp39-none-linux_aarch64.whl is not a supported wheel on this platform

### DISCUSSION THREAD

**Discussion by Scott8586** [mcguirepr89/BirdNET-Pi#847] [Q&A] [answered]
I'm trying to do the script based install - once it gets to installing the requirements file it stalls installing tflite_runtime:

`ERROR: tflite_runtime-2.6.0-cp39-none-linux_aarch64.whl is not a supported wheel on this platform`

The file exists in the directory - if I try to pip install it by hand, I get the same error.  I'm not sure where to go from here, and other answers to this question do not seem to apply or have not helped.

Thanks for any suggestions.

This is on a Pi4B+, the setup is:

Linux raspi42 6.1.21-v8+ #1642 SMP PREEMPT Mon Apr  3 17:24:16 BST 2023 aarch64 GNU/Linux

NAME="Raspbian GNU/Linux"
VERSION_ID="11"
VERSION="11 (bullseye)"
VERSION_CODENAME=bullseye

Python:
~/BirdNET-Pi/birdnet/bin/python3
Python 3.9.2 (default, Mar 12 2021, 04:06:34) 
[GCC 10.2.1 20210110] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> 
>>> import platform
>>> platform.architecture()
('32bit', 'ELF')

**Comment by jammbrose** [comment, marked as answer]
You are using a 32 bit platform.  The included version of Tensorflow is for 64 bit platforms.

**Comment by Scott8586** [comment]
Sorry - nevermind - I don't know I ended up with a 32bit python.  I installed a distro from scratch, and all is OK...

**Comment by buttim** [comment]
I have the very same problem but working on a 64bit platform.
`uname` says:
`Linux minivip 5.10.103-v8+ #1529 SMP PREEMPT Tue Mar 8 12:26:46 GMT 2022 aarch64 GNU/Linux`

**Comment by mfalkvidd** [comment]
Can confirm this problem on a (non-raspberry) arm64 platform. `Linux birdnet 6.1.26-05272-g26c406245a2c #1 SMP PREEMPT_DYNAMIC Thu Apr 27 10:15:40 UTC 2023 aarch64 aarch64 aarch64 GNU/Linux`

The board is a "Le potato": https://libre.computer/products/aml-s905x-cc/

I know Raspberry Pi is the only platform supported officially, but since it is almost impossible to source Raspberry Pi boards (for a long time now) I was hoping to set up my second and third stations using Le Potato.

```
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=22.10
DISTRIB_CODENAME=kinetic
DISTRIB_DESCRIPTION="Ubuntu 22.10
```

**Comment by ehpersonal38** [comment]
Please see this related issue, I commented a fix that I think should work if you scroll down

https://github.com/mcguirepr89/BirdNET-Pi/discussions/775

**Reply by mfalkvidd** [reply]
Great, thanks! I'll try it and report back.

**Comment by buttim** [comment]
In my case the output of the Python command you suggested (`platform.architecture()`) is
`('32bit', 'ELF')`
so this might actually be a 32bit system in spite of what `uname` reports maybe ?

**Reply by ehpersonal38** [reply]
That is possible, what is "Linux minivip"?

**Comment by buttim** [comment]
minivip is the hostname, it's a Raspberry PI 4

**Reply by ehpersonal38** [reply]
Hm, well your Python says 32 bit, so I'd suggest a re-image with the latest 64bit OS.

**Reply by mfalkvidd** [reply]
It is possible to install 32-bit python on a 64-bit linux. Not very common, but possible. Maybe that's what you are experiencing @buttim ? What source is your distribution/image from?

It might be possible to install the 64-bit version by running `sudo apt-get install python3:arm64`

**Reply by ehpersonal38** [reply]
That is interesting, I wonder what would cause the 32bit version to be installed instead of the 64bit.

**Reply by mfalkvidd** [reply]
Maybe building from source and selecting the 32-bit version instead of 64-bit?

**Comment by buttim** [comment]
I rebuilt Python from source code, the default version was 3.7, so no good for BirdNET

**Comment by thomas8240** [comment]
@ehpersonal38 
I am using the install instructions provided by Mcguirepr89. They are as follows for the Birdnet Pi:
https://www.raspberrypi.com/software/  download for windows using putty via port 22
	key inputs for the Raspberrypi imager are from the installation guide:
		https://github.com/mcguirepr89/BirdNET-Pi/wiki/Installation-Guide
		Once Linux is installed, the following curl command is entered.
			curl -s https://raw.githubusercontent.com/mcguirepr89/BirdNET-Pi/main/newinstaller.sh | bash
	Error Received: ERROR: tflite_runtime-2.6.0-cp39-none-linux_aarch64.whl is not a supported wheel on this platform.


I appear to be on the correct Python version 
      > birdnet@birdnetpi1:~/BirdNET-Pi$ ~/BirdNET-Pi/birdnet/bin/python3
        Python 3.9.2 (default, Mar 12 2021, 04:06:34)
Python source according to the log output is..... This is the same as when it ran successfully about a week ago when I had no errors.
       Establishing a python virtual environment
       + python3 -m venv birdnet
      + source ./birdnet/bin/activate

but the platform version is 32bit and not 64bit.
       >>> platform.architecture()
       ('32bit', 'ELF')

I could try to load Python 3.9.2 64bit before I run the curl command but it appears that the 32bit version is being picked up somewhere in that script that was provided. Therefore, my install of Python 64bit would then be written over by the Python 32bit with the script execution. I hesitate to debug the script since I am novice.

Thoughts?
thomas8240

**Comment by pddpauw** [comment]
The problem remains.

I have newly installed Pi4 64, and the script is installing a 64bit Python version.  Think there is a mismatch between the version in the tflite (3.9) and the automatically installed Python version (3.11).

_Python 3.11.2 (main, Mar 13 2023, 12:18:29) [GCC 12.2.0] on linux
('64bit', 'ELF')_

system: 
_Linux pibirdnet1 6.1.0-rpi4-rpi-v8 #1 SMP PREEMPT Debian 1:6.1.54-1+rpt2 (2023-10-05) aarch64 GNU/Linux_

ERROR: tflite_runtime-2.6.0-cp39-none-linux_aarch64.whl is not a supported wheel on this platform



So, if this is the problem indeed, how can we have the installation script referring to cp311 instead of cp 39?

**Reply by tphakala** [reply]
Did you install Bullseyey 64bit version on Raspberry Pi OS? Latest bookwork is not supported.

**Reply by pddpauw** [reply]
reinstalled with a legacy OS (bullseye) 64bit, but having the issue that a 32bit python version is installed.
Same error, but different rootcause.

system:
_Linux pibirdnet 6.1.21-v8+ #1642 SMP PREEMPT Mon Apr  3 17:24:16 BST 2023 aarch64 GNU/Linux_ 
debian version 11.8

**Reply by pddpauw** [reply]
managed to get it working with debian version 11.7!
So, for whomever experiencing issues: check to install a legacy 64bit OS, in my case 11.7 worked, but 11.8 not (wrong version of Python got installed on the 11.8).  Good luck!

**Comment by PragPossA** [comment]
I'm also getting the same error, but on a Raspberry Pi 5 . . . 
ERROR: tflite_runtime-2.6.0-cp39-none-linux_aarch64.whl is not a supported wheel on this platform.
 ~/BirdNET-Pi/birdnet/bin/python
Python 3.11.2 (main, Mar 13 2023, 12:18:29) [GCC 12.2.0] on linux
>>> import platform
>>> platform.architecture()
('64bit', 'ELF')

cat /etc/os-release  --> PRETTY_NAME="Debian GNU/Linux 12 (bookworm)"
cat /etc/debian_version --> 12.4

Does this work on Pi5 yet?

**Reply by pddpauw** [reply]
there are a couple of changes needed to make this work on a Pi5 .. look around in the discussions or issues, and you will find the redirect to a working version on Pi5

**Reply by PragPoss** [reply]
Thaw worked like a charm!  Thanks.

https://github.com/pddpauw/BirdPi

**Comment by krummrey** [comment]
I just ran into the same problem with a fresh install on my Pi4

> ERROR: tflite_runtime-2.6.0-cp39-none-linux_aarch64.whl is not a supported wheel on this platform.

but as far as I know I'm on a 64bit OS

> OS: Debian GNU/Linux 12 (bookworm) aarch64
> Host: Raspberry Pi 4 Model B Rev 1.5 
> Kernel: 6.1.0-rpi7-rpi-v8

The first check is passed. `uname -m` returns a `aarch64`
It fails when trying to install the requirements.txt for the python environment.

`pip3 install tflite_runtime-2.6.0-cp39-none-linux_aarch64.whl`
```
Looking in indexes: https://pypi.org/simple, https://www.piwheels.org/simple
ERROR: tflite_runtime-2.6.0-cp39-none-linux_aarch64.whl is not a supported wheel on this platform.
```

Maybe it is related to this: [piwheels does not currently support aarch64](https://github.com/piwheels/packages/issues/119)

Trying to compile TFlite [using this guide](https://qengineering.eu/install-tensorflow-2-lite-on-raspberry-pi-4.html)
Install the build tools: 
`sudo apt-get install cmake curl g++-arm-linux-gnueabihf`

download the latest TensorFlow version (2.6.0)
`wget -O tensorflow.zip https://github.com/tensorflow/tensorflow/archive/v2.6.0.zip`

unpack and give the folder a convenient name
`unzip tensorflow.zip `
`mv tensorflow-2.6.0 tensorflow`
`cd tensorflow `

get the dependencies
`./tensorflow/lite/tools/make/download_dependencies.sh`

run the C++ installation
`./tensorflow/lite/tools/make/build_rpi_lib.sh`

After that I commented out the wheel from the requirements.txt and ran the installer again.
It went through, but when I try connect to `birdnetpi.local` all I get is the Apache2 Debian Default Page. 

That's where I'm stuck now.

I ended up using this repo, which has been updated to work with the current version of Raspbian: [https://github.com/pddpauw/BirdPi](https://github.com/pddpauw/BirdPi)

**Reply by bsrotten** [reply]
Thanks the pddpauw repo worked for me too.

**Comment by mjforan** [comment]
Still broken 🫤
The "legacy" OS in the Imager app is now too recent as well. This image worked for me:
https://downloads.raspberrypi.com/raspios_lite_arm64/images/raspios_lite_arm64-2023-05-03/

**Reply by lukewalker2010** [reply]
This worked for me. Thank you!

**Reply by cweakland** [reply]
Did two installs today, worked like a charm, thanks!

**Reply by CamielBerkers** [reply]
Super! it did the job, thanks!

**Reply by joaldes** [reply]
This also worked for me.  Thanks!

**Reply by breimann** [reply]
@mjforan I just used your 2023-05-03 image and all I when I've ssh'd in is nothing when I enter the installation command:

> login as: birdnet1
> birdnet1@192.168.1.119's password:
> Linux birdnetpi1 6.1.21-v8+ #1642 SMP PREEMPT Mon Apr  3 17:24:16 BST 2023 aarch64
> 
> The programs included with the Debian GNU/Linux system are free software;
> the exact distribution terms for each program are described in the
> individual files in /usr/share/doc/*/copyright.
> 
> Debian GNU/Linux comes with ABSOLUTELY NO WARRANTY, to the extent
> permitted by applicable law.
> birdnet1@birdnetpi1:~ $ curl -s https://raw.githubusercontent.com/mcguirepr89/BirdNET-Pi/main/newinstaller.sh | bash
> birdnet1@birdnetpi1:~ $
> 

It does completely.... nothing!!! I've spent so many hours on this, please help!

I'm using a rPi4 and a 64gig micro sd card

I used this image:

https://downloads.raspberrypi.com/raspios_lite_arm64/images/raspios_lite_arm64-2023-05-03/2023-05-03-raspios-bullseye-arm64-lite.img.xz

**Reply by tphakala** [reply]
This app has been abandoned and has not been maintained in many years anymore. Please switch to either https://github.com/tphakala/birdnet-go or https://github.com/Nachtzuster/BirdNET-Pi

**Comment by birdboyer** [comment]
The legacy OS on the Imager app that works is "Raspberry Pi OS (Legacy, 64-bit) Lite - A port of Debian Bullseye with security updates and no desk top environment - released: 2024-03-12 - on line 0.3 GB download". It's the next to the last one in the dropdown menu after selecting "Raspberry Pi OS (other), 4th down in the main menu. I tried all the other (Legacy, 64-bit) versions and they all gagged at the error message referenced in the heading. But this one installed successfully.

**Comment by breimann** [comment]
Thankyou tomi

**Comment by brunnwart** [comment]
Yes, thanks @birdboyer:
- Raspberry Pi OS Lite (64 bit) (still) leads to the error in this thread whereas
- Raspberry Pi OS (Legacy, 64-bit) Lite installation works well
Maybe someone could update the installation guide before everybody runs into the same error?

**Reply by FaberFlorian** [reply]
https://github.com/Nachtzuster/BirdNET-Pi
